Registry/RegistryParse.py

```diff
@@ -1426,3 +1518,242 @@ def records(self):
+        should be the REGFBlock, and the parents of other HvLEBlocks should be the preceeding
+        HvLEBlocks.
+        """
+        super(HvLEBlock, self).__init__(buf, offset, parent)
+
+        _id = self.unpack_dword(0)
+        if _id != 0x454C7648:
+            raise ParseException("Invalid HvLE ID")
+
+        self._offset_next_hvle = self._offset + self.size()
+        self._marvin32seed = 0x82EF4D887A4E55C5
+
+    def __str__(self):
+        return "HvLE at 0x%x" % (self._offset)
+
+    def marvin32_hash(self, buf):
+        """
+        Hash the buf using Marvin32 with a predefined seed.
+        """
+        def rotl(x, n, w):
+            return (x.value << n) | (x.value >> (w - n))
+
+        def to_uint32_le(four_bytes):
+            b1, b2, b3, b4 = bytearray(four_bytes)
+            return b1 | (b2 << 8) | (b3 << 16) | (b4 << 24)
+
+        def marvin32_mix(state, val):
+            lo, hi = state
+            lo.value += val.value
+            hi.value ^= lo.value
+            lo.value = rotl(lo, 20, 32) + hi.value
+            hi.value = rotl(hi, 9, 32) ^ lo.value
+            lo.value = rotl(lo, 27, 32) + hi.value
+            hi.value = rotl(hi, 19, 32)
+            return (lo, hi)
+
+        seed = self._marvin32seed
+        lo = c_uint32(seed)
+        hi = c_uint32(seed >> 32)
+        state = (lo, hi)
+
+        length = len(buf)
+        pos = 0
+        val = c_uint32()
+
+        while length >= 4:
+                val.value = to_uint32_le(buf[pos:pos+4])
+                state = marvin32_mix(state, val)
+                pos += 4
+                length -= 4
+
+        final = c_uint32(0x80)
+        if length == 3:
+                final.value = (final.value << 8) | buf[pos+2]
+        elif length == 2:
+            final.value = (final.value << 8) | buf[pos+1]
+        elif length == 1:
+            final.value = (final.value << 8) | buf[pos]
+
+        state = marvin32_mix(state, final)
+        state = marvin32_mix(state, c_uint32(0))
+        lo, hi = state
+        return (hi.value << 32 | lo.value)
+
+    def size(self):
+        """
+        Get the size of this HvLEBlock.
+        """
+        return self.unpack_dword(0x4)
+
+    def hash_1(self):
+        """
+        Get the value of Hash-1.
+        """
+        return self.unpack_qword(0x18)
+
+    def calculate_hash_1(self):
+        """
+        Calculate the Hash-1.
+        """
+        return self.marvin32_hash(self._buf[self._offset+40:self._offset+self.size()])
+
+    def hash_2(self):
+        """
+        Get the value of Hash-2.
+        """
+        return self.unpack_qword(0x20)
+
+    def calculate_hash_2(self):
+        """
+        Calculate the Hash-2.
+        """
+        return self.marvin32_hash(self._buf[self._offset:self._offset+32])
+
+    def validate_log_entry(self):
+        """
+        Check if this log entry is valid.
+        """
+        if (self.size() <= 40) or (self.size() % 0x200 != 0):
```

Comment:
These are not magic numbers. Or shall I define something like `LOG_ENTRY_SIZE_MIN` and `LOG_ENTRY_SIZE_ALIGNMENT`?